stable/pomerium/templates/tls-secrets.yaml

```diff
@@ -16,3 +16,129 @@ data:
 ---
 {{- end }}
 {{- end }}
+
+{{- define "pomerium.authenticate.tlsSecretObject" -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    app.kubernetes.io/name: {{ template "pomerium.name" . }}
+    helm.sh/chart: {{ template "pomerium.chart" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+  annotations:
+{{- if .Values.config.generateTLS }}
+    helm.sh/hook: pre-install
+{{- end }}
+  name: {{ template "pomerium.authenticate.tlsSecret.name" . }}
+  namespace: {{ .Release.Namespace }}
+type: kubernetes.io/tls
+data:
+{{- end -}}
+
+{{- define "pomerium.authorize.tlsSecretObject" -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    app.kubernetes.io/name: {{ template "pomerium.name" . }}
+    helm.sh/chart: {{ template "pomerium.chart" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+  annotations:
+{{- if .Values.config.generateTLS }}
+    helm.sh/hook: pre-install
+{{- end }}
+  name: {{ template "pomerium.authorize.tlsSecret.name" . }}
+  namespace: {{ .Release.Namespace }}
+type: kubernetes.io/tls
+data:
+{{- end -}}
+
+{{- define "pomerium.proxy.tlsSecretObject" -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    app.kubernetes.io/name: {{ template "pomerium.name" . }}
+    helm.sh/chart: {{ template "pomerium.chart" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+  annotations:
+{{- if .Values.config.generateTLS }}
+    helm.sh/hook: pre-install
+{{- end }}
+  name: {{ template "pomerium.proxy.tlsSecret.name" . }}
+  namespace: {{ .Release.Namespace }}
+type: kubernetes.io/tls
+data:
+{{- end -}}
+
+{{- define "pomerium.ca.tlsSecretObject" -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    app.kubernetes.io/name: {{ template "pomerium.name" . }}
+    helm.sh/chart: {{ template "pomerium.chart" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+  annotations:
+{{- if .Values.config.generateTLS }}
+    helm.sh/hook: pre-install
```

Comment:
@travisgroth while looking at helm doc, I wonder if there is not some trick with the way hooks work, see https://github.com/helm/helm/blob/master/docs/charts_hooks.md#automatically-delete-hook-from-previous-release

For example, once the release has been installed, if we change the chart version say, this will update the secret, will this work as expected? Sorry I don't have the bandwidth to test all those things right now and find answers :/

Reply:
No worries.  We're relying on IsInstall check to only run this upon initial install.  This means the hook resources only get created once and then will not run again.  We're intentionally "abandoning" the resources by leaving out the lifecycle hook since we want them to stick around.

This is the only way to autogenerate once that I could figure out.

Reply:
indeed, it's not easy at all :) this should be sufficient, and we will get feedbacks from users if there are cases we missed :)

Reply:
🤞